RFC-002-SWAP.md

```diff
@@ -10,267 +10,273 @@
+Such an exchange can thus be described as
+- transferring the ownership of the **alpha-asset** on the **alpha-ledger** from party A to party B
+- transferring the ownership of the **beta-asset** on the **beta-ledger** from party B to party A
 
-### `headers`
+This terminology has several advantages:
 
-#### `alpha_ledger`
-```
-Type: Ledger 
-```
+1. it does not imply an order (compared to a terminology like **first**/**second**)
+2. it is not subjective to any of the parties (compared to a terminology like **source**/**target**, **incoming**/**outgoing** or **local**/**remote**)
 
-The ledger on which the Sender sells and the Receiver buys[²].
+## BAM! messages
 
-#### `beta_ledger`
-```
-Type: Ledger
-```
-The ledger on which the Sender buys and the Receiver sells.
+The messages to negotiate such an exchange consist of a `REQUEST` frame and an according `RESPONSE` frame.
 
-#### `alpha_asset`
-```
-Type: Asset
-```
+### SWAP REQUEST frame
 
-The asset that the Sender sells and the Receiver buys.
+#### Definition
 
-#### `beta_asset`
+A SWAP REQUEST message is a `FRAME` of type `REQUEST`.
+[As per definition](./RFC-001-BAM.md#type-1) in `BAM!`, a `REQUEST` `FRAME` has a `type` that defines its semantics.
+For the SWAP REQUEST message, this type is `SWAP`.
 
-The asset that the Sender buys and the Receiver sells.
+#### Headers
 
-#### `protocol`
-A protocol that defines the steps, transactions and communications needed to proceed with an asset exchange.
+To express all the information for an exchange, a SWAP REQUEST MUST include the following headers:
 
-The exact value of this header will be defined in subsequent swap protocol RFCs.
+- `alpha_ledger`: describes the ledger the alpha-asset is tracked on
+- `beta_ledger`: describes the ledger the beta-asset is tracked on
+- `alpha_asset`: describes the asset who's ownership is transferred on the alpha-ledger
+- `beta_asset`: describes the asset who's ownership is transferred on the beta-ledger
+- `protocol`: describes the protocol that is used to transfer the ownership of the assets
 
-### `body`
-Refer to the RFC of the given swap protocol.
+This RFC only defines these headers.
+The actual definition of ledgers, assets and protocols is not subject to this RFC.
 
-## SWAP RESPONSE
+#### Body
 
-### Definition
-The SWAP RESPONSE message is a `FRAME` of type `RESPONSE`.
+The body of a SWAP REQUEST depends on the chosen `protocol`.
+It is therefore subject to RFCs which define such protocols to also define, which information is contained in the `body` of a SWAP REQUEST/RESPONSE frame.
 
-### `status`
+### SWAP RESPONSE frame
 
-See [RFC-001](./RFC-001-BAM.md#status-code-families) for more details, including the definition of statuses `XX00-19`.
+#### Definition
 
-RFC-002 reserves statuses 20 to 39 across all families.
-Protocols may define the meaning of statuses 40 and above.
+A frame of type `REQUEST` implies a `RESPONSE`.
+We will refer to the response of a `SWAP REQUEST` as SWAP RESPONSE.
```

Comment:
:wrench: Only place where swap request or swap response is highlighted this way.
```suggestion
We will refer to the response of a SWAP REQUEST as SWAP RESPONSE.
```